### PR TITLE
Make operator deployable via OLM

### PR DESCRIPTION
**Pull request by sebsoto**
This PR consists of 3 commits which work towards making the WMCO deployable via OLM:
* Change the WNI resource tracking directory used so that the operator can be deployed outside of CI
* Change RBAC permissions so the SA used by the operator has the correct permissions
* Add the operator bundle and olm deployment yamls

### CODE REVIEW

File: deploy/bundle/catalogsource.yaml [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+        name: windows-machine-config-operator.v0.0.1
+        namespace: placeholder
+        annotations:
+          alm-examples: >-
+            [{"apiVersion":"","kind":"WindowsMachineConfig","metadata":{"name":""},"spec":{}}]
+          categories: OpenShift Optional
+          certified: 'false'
+          createdAt: ''
+          description: >-
+            This operator allows for the addition of Windows nodes to an OpenShift
```
Comment (ravisantoshgudimetla) [inline deploy/bundle/catalogsource.yaml]: management of the Windows Nodes in an OpenShift cluster

File: deploy/bundle/catalogsource.yaml [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        customresourcedefinitions:
+          owned:
+            - name: windowsmachineconfigs.wmc.openshift.io
+              displayName: WindowsMachineConfig
+              kind: WindowsMachineConfig
+              version: v1alpha1
+              description: Windows Machine Config
+              resources:
+                - version: v1
+                  kind: Deployment
```
Comment (ravisantoshgudimetla) [inline deploy/bundle/catalogsource.yaml]: Not sure where you're getting all these resources from?

File: deploy/bundle/catalogsource.yaml [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        scope: Namespaced
+        subresources:
+          status: {}
+        validation:
+          openAPIV3Schema:
+            description: WindowsMachineConfig is the Schema for the windowsmachineconfigs API
+            properties:
+              apiVersion:
+                description: >-
+                  APIVersion defines the versioned schema of this representation of an
```
Comment (ravisantoshgudimetla) [inline deploy/bundle/catalogsource.yaml]: I think you need to update this to talk about our api?

File: deploy/bundle/catalogsource.yaml [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+                  object. Servers should convert recognized schemas to the latest
+                  internal value, and may reject unrecognized values. More info:
+                  https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+                type: string
+              kind:
+                description: >-
+                  Kind is a string value representing the REST resource this object
```
Comment (ravisantoshgudimetla) [inline deploy/bundle/catalogsource.yaml]: Same as above?

File: deploy/bundle/README.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+This directory contains resources related to installing the WMCO onto a cluster using OLM.
```
Comment (ravisantoshgudimetla) [inline deploy/bundle/README.md]: I think you also need to mention  you assume OLM has been installed.

File: deploy/bundle/catalogsource.yaml [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+            storage: true
+  packages: |
+    - channels:
+      - currentCSV: windows-machine-config-operator.v0.0.1
+        name: alpha
+      defaultChannel: "alpha"
+      packageName: windows-machine-config-operator
+kind: ConfigMap
+metadata:
+  name: wmco-operators
```
Comment (ravisantoshgudimetla) [inline deploy/bundle/catalogsource.yaml]: why this name?

File: deploy/bundle/catalogsource.yaml (hunk 3 above)
Comment (sebsoto) [reply deploy/bundle/catalogsource.yaml]: Im pretty sure this just describes the apiVersion field

File: deploy/bundle/catalogsource.yaml (hunk 4 above)
Comment (sebsoto) [reply deploy/bundle/catalogsource.yaml]: same as above but for kind

File: deploy/bundle/catalogsource.yaml (hunk 6 above)
Comment (sebsoto) [reply deploy/bundle/catalogsource.yaml]: Its an arbitrary name. Whats wrong with it?

File: deploy/bundle/catalogsource.yaml (hunk 2 above)
Comment (sebsoto) [reply deploy/bundle/catalogsource.yaml]: From deploying the operator and seeing what it created. I need to remove secret and configmap

File: deploy/bundle/catalogsource.yaml (hunk 6 above)
Comment (sebsoto) [reply deploy/bundle/catalogsource.yaml]: Makes sense, I'll change it to wmc-operators

File: deploy/bundle/README.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
+# Bundling the Windows Machine Config Operator
```
Comment (aravindhp) [inline deploy/bundle/README.md]: Please add this to the main README.

File: deploy/bundle/README.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+This directory contains resources related to installing the WMCO onto a cluster using OLM.
+
+## Pre-requisites
+This guide assumes [OLM](https://github.com/operator-framework/operator-lifecycle-manager) has been installed to the
```
Comment (aravindhp) [inline deploy/bundle/README.md]: We are targeted at OpenShift 4.x and OLM is part of it.

File: deploy/bundle/README.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+cluster, and [operator-sdk](https://github.com/operator-framework/operator-sdk) and
+[opm](https://github.com/operator-framework/operator-registry/) have been installed on localhost.
```
Comment (aravindhp) [inline deploy/bundle/README.md]: `on the localhost`

File: deploy/bundle/README.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+## Creating a bundle image
+A bundle image can be created by editing the CSV in deploy/bundle/windows-machine-config-operator/manifests/
+and replacing `$OPERATOR_IMAGE` with the location of the operator image you wish to deploy.
```
Comment (aravindhp) [inline deploy/bundle/README.md]: WMCO operator image. Link to the build section of the README.

File: deploy/bundle/README.md [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+```
+
+### Editing an existing operator index
+An existing operator index can have bundles added to it:
+```
+opm index add --from-index $INDEX_VERSION
+```
+and removed from it:
+```
+opm index rm
```
Comment (aravindhp) [inline deploy/bundle/README.md]: Don't we need to specify the index version?

File: deploy/bundle/README.md [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+oc apply -f catalogsource.yaml
+```
+This will deploy a CatalogSource object in your current namespace. You can check the status of it via:
+```
+oc describe catalogsource
+```
+Now wait 1-10 minutes for the catalogsource's status.connectionState.lastObservedState field to be set to READY.
+Once this is done, log into the cluster's web console, then go to the OpenShift GUI console. In 'Administration' mode,
+go to the Operators/OperatorHub tab. Search for the 'Windows Machine Config Operator' and click install.
+The operator will then be installed onto your cluster.
```
Comment (aravindhp) [inline deploy/bundle/README.md]: Given you have been doing everything in the command line, please do the same here rather than using the GUI. This will also help with the follow up testing story when a script would need to be written. You can have the GUI section as an alternate deployment section. 

I am remember the bundle needing a `package.yaml`. Has that requirement been removed?

File: deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml [hunk 13]
```diff
@@ -0,0 +1,181 @@
...
+spec:
+  displayName: Windows Machine Config
+  description: |
+    This operator allows for the management of Windows nodes in an OpenShift
+    cluster.
+  maturity: alpha
+  version: 0.0.1
+  replaces: ''
+  skips: []
+  minKubeVersion: ''
```
Comment (aravindhp) [inline deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml]: Please fill this in with whatever version 4.5 ships with.

File: deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml [hunk 14]
```diff
@@ -0,0 +1,181 @@
...
+          rules:
+          - apiGroups:
+            - ""
+            resources:
+            - pods
+            - services
+            - services/finalizers
+            - endpoints
+            - persistentvolumeclaims
+            - events
```
Comment (aravindhp) [inline deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml]: Are we creating these resources? Only resources that we are actually using should be added here.

File: deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml [hunk 15]
```diff
@@ -0,0 +1,181 @@
...
+            - configmaps
+            - secrets
+            verbs:
+            - create
+            - delete
+            - get
+            - list
+            - patch
```
Comment (aravindhp) [inline deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml]: Are we using patch?

File: deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml [hunk 16]
```diff
@@ -0,0 +1,181 @@
...
+            - update
+            - watch
+          - apiGroups:
+            - apps
+            resources:
+            - deployments
+            - daemonsets
+            - replicasets
+            - statefulsets
```
Comment (aravindhp) [inline deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml]: Are we creating these resources?

File: deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml [hunk 17]
```diff
@@ -0,0 +1,181 @@
...
+            - get
+            - create
+          - apiGroups:
+            - apps
+            resourceNames:
+            - windows-machine-config-operator
+            resources:
+            - deployments/finalizers
+            verbs:
+            - update
```
Comment (aravindhp) [inline deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml]: Why is this needed?

File: deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml [hunk 18]
```diff
@@ -0,0 +1,181 @@
...
+            - pods
+            verbs:
+            - get
+          - apiGroups:
+            - apps
+            resources:
+            - replicasets
+            - deployments
+            verbs:
+            - get
```
Comment (aravindhp) [inline deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml]: This seems like a repeat.

File: deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml [hunk 19]
```diff
@@ -0,0 +1,181 @@
...
+          - version: v1
+            kind: Pod
+        specDescriptors: []
+        statusDescriptors: []
+    required: []
+  install:
+    strategy: deployment
+    spec:
+      permissions:
+        - serviceAccountName: windows-machine-config-operator
```
Comment (aravindhp) [inline deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml]: I thought this could go into a separate SA yaml and referenced here. Please confirm if this is possible with the OLM team.

File: deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml [hunk 20]
```diff
@@ -0,0 +1,181 @@
...
+            resources:
+            - '*'
+            verbs:
+            - create
+            - delete
+            - get
+            - list
+            - patch
+            - update
+            - watch
```
Comment (aravindhp) [inline deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml]: Isn't this equivalent to `*`?

File: deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml [hunk 21]
```diff
@@ -0,0 +1,181 @@
...
+                serviceAccountName: windows-machine-config-operator
+  installModes:
+    - type: OwnNamespace
+      supported: true
+    - type: SingleNamespace
+      supported: true
+    - type: MultiNamespace
+      supported: false
+    - type: AllNamespaces
+      supported: false
```
Comment (aravindhp) [inline deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml]: We don't need to support all these install modes. We just need the mode that allows us to watch our own namespace.

File: deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml [hunk 22]
```diff
@@ -0,0 +1,181 @@
...
+            - delete
+            - get
+            - list
+            - patch
+            - update
+            - watch
+      deployments:
+        - name: windows-machine-config-operator
+          spec:
+            replicas: 1
```
Comment (aravindhp) [inline deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml]: We might want more than one replica.

File: deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml [hunk 23]
```diff
@@ -0,0 +1,181 @@
...
+            - '*'
+            verbs:
+            - create
+            - delete
+            - get
+            - list
+            - patch
+            - update
+            - watch
+      deployments:
```
Comment (aravindhp) [inline deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml]: I know there was some talk about being use a separate YAML and referencing it. Please check with the OLM team.

File: deploy/bundle/README.md (hunk 12 above)
Comment (sebsoto) [reply deploy/bundle/README.md]: I wasn't aware you could do it via CLI. I'll figure out how.

The package.yaml requirement has been removed. Heres an example of the new way to bundle operators: https://github.com/operator-framework/bundle-example/tree/master/bundle/manifests

File: deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml (hunk 14 above)
Comment (sebsoto) [reply deploy/bundle/windows-machine-config-operator/manifests/windows-machine-config-operator.v0.0.1.clusterserviceversion.yaml]: I copied these from whats in /deploy. I can trim it down

File: README.md [hunk 24]
```diff
@@ -30,3 +30,119 @@ Once the above variables are set:
...
+You should use this new directory when [creating the bundle image](#creating-a-bundle-image)
+
+### Creating a bundle image
+A bundle image can be created by editing the CSV in deploy/bundle/windows-machine-config-operator/manifests/
+and replacing `REPLACE_IMAGE` with the location of the WMCO operator image you wish to deploy.
+See [the build instructions](#build) for more information on building the image.
+
+You can then run the following command in the root of this git repository
+```shell script
+operator-sdk bundle create $BUNDLE_REPOSITORY:$VERSION_TAG -d deploy/bundle/windows-machine-config-operator/manifests \
```
Comment (sebsoto) [inline README.md]: This command needs to updated to look like operator-sdk bundle create quay.io/ssoto/test:bundlev0.0.0 -d deploy/olm-catalog/windows-machine-config-operator/0.0.0 --channels alpha --default-channel alpha --image-builder podman
will do that after reviews

File: deploy/olm-catalog/windows-machine-config-operator/0.0.0/windows-machine-config-operator.v0.0.0.clusterserviceversion.yaml [hunk 25]
```diff
@@ -0,0 +1,194 @@
...
+                  secretName: cloud-private-key
+              - name: kubeconfig
+                secret:
+                  secretName: kubeconfig
+      permissions:
+      - rules:
+        - apiGroups:
+          - ""
+          resources:
+          - pods
```
[the remark this answers is not in the document]
Comment (sebsoto) [reply deploy/olm-catalog/windows-machine-config-operator/0.0.0/windows-machine-config-operator.v0.0.0.clusterserviceversion.yaml]: This file is generated from the files in `deploy/`

File: README.md [hunk 26]
```diff
@@ -30,3 +30,119 @@ Once the above variables are set:
...
+Put the kubeconfig you wish to use in a secret:
+```shell script
+# Change paths as necessary
+oc create secret generic kubeconfig --from-file=kubeconfig=/path/to/kubeconfig
+```
+
+Put the cluster address in a secret:
+```shell script
+CLUSTER_ADDR=$(oc cluster-info | head -n1 | sed 's/.*\/\/api.//g'| sed 's/:.*//g')
+oc create secret generic cluster-address --from-literal=cluster-address=$CLUSTER_ADDR
```
[the remark this answers is not in the document]
Comment (sebsoto) [reply README.md]: This was the easiest way I found to provide a variable environment variable.

File: README.md [hunk 27]
```diff
@@ -30,3 +30,119 @@ Once the above variables are set:
...
+```shell script
+# Change paths as necessary
+oc create secret generic cloud-credentials --from-file=credentials=/home/$user/.aws/credentials
+oc create secret generic cloud-private-key --from-file=private-key.pem=/home/$user/.ssh/$keyname
+```
+
+Put the kubeconfig you wish to use in a secret:
+```shell script
+# Change paths as necessary
+oc create secret generic kubeconfig --from-file=kubeconfig=/path/to/kubeconfig
```
[the remark this answers is not in the document]
Comment (sebsoto) [reply README.md]: The user can generate the kubeconfig from the sa if they wish, I did not document that process as it isn't necessary and will become uneeded once WINC-328 is done

[the remark this answers is not in the document]
Comment (sebsoto) [reply README.md]: I can explicitly call it out if you want

[the remark this answers is not in the document]
Comment (sebsoto) [reply README.md]: Done

File: pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go [hunk 28]
```diff
@@ -144,7 +144,7 @@ func (r *ReconcileWindowsMachineConfig) Reconcile(request reconcile.Request) (re
 			// We assume the credential path is `/etc/aws/credentials` mounted as a secret.
 			wkl.CloudCredentialsPath,
 			instance.Spec.AWS.CredentialAccountID,
-			"", "", instance.Spec.InstanceType,
+			"/tmp", "", instance.Spec.InstanceType,
```
Comment (aravindhp) [inline pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go]: Is it guaranteed that `/tmp` always exists?

File: README.md [hunk 29]
```diff
@@ -45,3 +45,120 @@ Once the above variables are set and the /payload directory has been populated,
 hack/run-ci-e2e-test.sh
 ```
 
+## Bundling the Windows Machine Config Operator
+This directory contains resources related to installing the WMCO onto a cluster using OLM.
+
+### Pre-requisites
+[operator-sdk](https://github.com/operator-framework/operator-sdk) and
```
Comment (aravindhp) [inline README.md]: Please link to the operator pre-reqs

File: README.md [hunk 30]
```diff
@@ -45,3 +45,120 @@ Once the above variables are set and the /payload directory has been populated,
...
+[opm](https://github.com/operator-framework/operator-registry/) have been installed on the localhost.
+
+### Generating a new bundle
+If changes need to be made to the bundle spec, a new bundle can be generated with:
```
Comment (aravindhp) [inline README.md]: This leads me to ask what do I need to do if I am starting from scratch. Do I not need to run this step?

File: deploy/olm-catalog/windows-machine-config-operator/0.0.0/windows-machine-config-operator.v0.0.0.clusterserviceversion.yaml (hunk 25 above)
[the remark this answers is not in the document]
Comment (ravisantoshgudimetla) [reply deploy/olm-catalog/windows-machine-config-operator/0.0.0/windows-machine-config-operator.v0.0.0.clusterserviceversion.yaml]: Which file in  `deploy` should we change?

[the remark this answers is not in the document]
Comment (sebsoto) [reply deploy/olm-catalog/windows-machine-config-operator/0.0.0/windows-machine-config-operator.v0.0.0.clusterserviceversion.yaml]: The permissions are generated from `deploy/role.yaml`

[the remark this answers is not in the document]
Comment (aravindhp) [reply deploy/olm-catalog/windows-machine-config-operator/0.0.0/windows-machine-config-operator.v0.0.0.clusterserviceversion.yaml]: There is something amiss here. Do these need to be changed after the generation? Or do the `deploy/` files need to be fixed?

File: deploy/olm-catalog/windows-machine-config-operator/0.0.0/windows-machine-config-operator.v0.0.0.clusterserviceversion.yaml [hunk 31]
```diff
@@ -0,0 +1,194 @@
...
+    strategy: deployment
+  installModes:
+  - supported: true
+    type: OwnNamespace
+  - supported: true
+    type: SingleNamespace
+  - supported: false
+    type: MultiNamespace
+  - supported: true
+    type: AllNamespaces
```
Comment (aravindhp) [inline deploy/olm-catalog/windows-machine-config-operator/0.0.0/windows-machine-config-operator.v0.0.0.clusterserviceversion.yaml]: Why do we support this?

File: deploy/olm-catalog/windows-machine-config-operator/0.0.0/windows-machine-config-operator.v0.0.0.clusterserviceversion.yaml (hunk 25 above)
[the remark this answers is not in the document]
Comment (sebsoto) [reply deploy/olm-catalog/windows-machine-config-operator/0.0.0/windows-machine-config-operator.v0.0.0.clusterserviceversion.yaml]: The original generation of the repo `operator-sdk new` created the depoly files. We are free to change them now. Those deploy files are used to generate the csv with `operator-sdk generate csv`. Its up to us to manually change the deploy files, and rerun the csv generation to change this file.

File: pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go (hunk 28 above)
Comment (sebsoto) [reply pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go]: As long as we control the image creation, yes.

File: README.md (hunk 30 above)
Comment (sebsoto) [reply README.md]: You wont be starting from "scratch", as the bundle i've seeded here now exists. We just need to update the bundle with the generate command in the case that we change the yaml files. I've updated the PR to explain that.

I've also created https://issues.redhat.com/browse/WINC-332 to ensure that the bundle is being updated when necessary.

File: deploy/olm-catalog/windows-machine-config-operator/0.0.0/windows-machine-config-operator.v0.0.0.clusterserviceversion.yaml (hunk 31 above)
Comment (sebsoto) [reply deploy/olm-catalog/windows-machine-config-operator/0.0.0/windows-machine-config-operator.v0.0.0.clusterserviceversion.yaml]: We shouldnt, changed.

File: deploy/olm-catalog/windows-machine-config-operator/0.0.0/windows-machine-config-operator.v0.0.0.clusterserviceversion.yaml (hunk 25 above)
[the remark this answers is not in the document]
Comment (aravindhp) [reply deploy/olm-catalog/windows-machine-config-operator/0.0.0/windows-machine-config-operator.v0.0.0.clusterserviceversion.yaml]: So where is that going to happen?

File: README.md [hunk 32]
```diff
@@ -45,3 +45,123 @@ Once the above variables are set and the /payload directory has been populated,
...
+
+Create the cloud provider and cloud private key secrets. The cloud-private-key should match the keypair used in the
+Windows Machine Config CR you will use.
+```shell script
+# Change paths as necessary
+oc create secret generic cloud-credentials --from-file=credentials=/home/$user/.aws/credentials
+oc create secret generic cloud-private-key --from-file=private-key.pem=/home/$user/.ssh/$keyname
+```
+
+Put the kubeconfig you wish to use in a secret. In order to use the permissions of the windows-machine-config-operator
```
Comment (ravisantoshgudimetla) [inline README.md]: So the service account will be generated at a later [step](https://github.com/openshift/windows-machine-config-operator/pull/17/files#diff-04c6e90faac2675aa89e2176d2eec7d8R166) as part of csv creation whereas we want to generate the secret here with sa permissions here?

File: deploy/olm-catalog/windows-machine-config-operator/0.0.0/windows-machine-config-operator.v0.0.0.clusterserviceversion.yaml (hunk 25 above)
[the remark this answers is not in the document]
Comment (ravisantoshgudimetla) [reply deploy/olm-catalog/windows-machine-config-operator/0.0.0/windows-machine-config-operator.v0.0.0.clusterserviceversion.yaml]: So, if we already have the `KUBECONFIG` secret mounted, why do we need SA permissions?

File: deploy/operator.yaml [hunk 33]
```diff
@@ -31,3 +44,25 @@ spec:
...
+              valueFrom:
+                secretKeyRef:
+                  name: cluster-address
+                  key: cluster-address
+          volumeMounts:
+          # TODO: Remove the cloud-credentials and cloud-private-key volumeMounts as part of
+          # https://issues.redhat.com/browse/WINC-325
+          - name: cloud-credentials
+            mountPath: "/etc/cloud/"
+            readOnly: true
```
Comment (ravisantoshgudimetla) [inline deploy/operator.yaml]: Good job on making them readOnly

File: pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go (hunk 28 above)
Comment (ravisantoshgudimetla) [reply pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go]: It also depends on where we are copying the operator binary to as the trackerfile creation happens at the place where the binary is getting copied to. That being said, we should be a bit cautious where we're writing this file as this can help in debugging purposes.(We are also having configmap which we can track though but it won't have information on the sg and ports being opened etc)

File: README.md (hunk 32 above)
Comment (sebsoto) [reply README.md]: Yes the service account is defined in the `permissions` section of the csv which is generated

File: deploy/role.yaml [hunk 34]
```diff
@@ -7,37 +7,29 @@ rules:
...
+  - secrets
+  verbs:
+  - create
+  - delete
+  - get
+- apiGroups:
+  - ""
+  resources:
   - services
   - services/finalizers
```
[the remark this answers is not in the document]
Comment (sebsoto) [reply deploy/role.yaml]: Yes the metrics service requires it

File: deploy/role.yaml [hunk 35]
```diff
@@ -59,6 +51,8 @@ rules:
   - pods
```
[the remark this answers is not in the document]
Comment (sebsoto) [reply deploy/role.yaml]: Without the pod permission you get errors such as 
```
E0330 18:08:36.689296       1 reflector.go:280] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:222: Failed to watch *v1.Pod: unknown (get pods)
```

File: deploy/role.yaml [hunk 36]
```diff
@@ -59,6 +51,8 @@ rules:
...
   verbs:
   - get
+  - list
+  - watch
 - apiGroups:
   - apps
   resources:
```
[the remark this answers is not in the document]
Comment (sebsoto) [reply deploy/role.yaml]: Needed for metrics:
```
{"level":"info","ts":1585602600.380326,"logger":"cmd","msg":"Could not create metrics Service","error":"failed to initialize service object for metrics: replicasets.apps \"windows-machine-config-operator-69f9ccc79c\" is forbidden: User \"system:serviceaccount:windows-machine-config-operator:windows-machine-config-operator\" cannot get resource \"replicasets\" in API group \"apps\" in the namespace \"windows-machine-config-operator\""}
```

File: deploy/role.yaml (hunk 34 above)
[the remark this answers is not in the document]
Comment (aravindhp) [reply deploy/role.yaml]: Please add comment mentioning  that.

File: deploy/role.yaml (hunk 35 above)
[the remark this answers is not in the document]
Comment (aravindhp) [reply deploy/role.yaml]: This is a bug in our code. We are unnecessarily [watching](https://github.com/openshift/windows-machine-config-operator/blob/master/pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go#L76) pods. Please open a bug to track this and we should remove this from here when that bug is fixed.

File: deploy/role.yaml (hunk 36 above)
[the remark this answers is not in the document]
Comment (aravindhp) [reply deploy/role.yaml]: Please mention that here.
